### PR TITLE
reduced latency

### PR DESCRIPTION
**Pull request by konstantin-lysenko-netapp**

### CODE REVIEW

File: cmd/rollout_get.go [hunk 1]
```diff
@@ -73,21 +73,17 @@ func runRolloutGetAction(args []string) {
...
 		if rolloutGetOptions.TimeoutSeconds > 0 {
 			ts := time.Duration(rolloutGetOptions.TimeoutSeconds)
 			ctx, cancel = context.WithTimeout(ctx, ts*time.Second)
 			defer cancel()
 		}
-		stopCh := ctx.Done()
-		go controller.WatchRollout(stopCh, rolloutUpdates)
-		controller.Run(ctx)
-		close(rolloutUpdates)
+		wg := &sync.WaitGroup{}
```
Comment (caduri) [inline cmd/rollout_get.go:84]: add an empty line above

File: viewcontroller/viewcontroller.go [hunk 2]
```diff
@@ -108,48 +100,61 @@ func NewRolloutViewController(rolloutId string, noColor bool) *RolloutViewContro
 	}
 }
 
-func (c *RolloutViewController) GetRollout() (rollout.DetailedRollout, error) {
-	detailedRollout := rollout.DetailedRollout{}
+func (c *RolloutViewController) GetRollout() (*rollout.DetailedRollout, error) {
+	detailedRollout := &rollout.DetailedRollout{}
+	errors := make(chan error)
+	wg := &sync.WaitGroup{}
+	wg.Add(3)
```
Comment (caduri) [inline viewcontroller/viewcontroller.go:107]: what does 3 means?

File: viewcontroller/viewcontroller.go [hunk 3]
```diff
@@ -108,48 +100,61 @@ func NewRolloutViewController(rolloutId string, noColor bool) *RolloutViewContro
...
 
 	if c.previousRollout == nil {
-		strategy, err := c.rolloutRepository.GetStrategy(c.rolloutId)
-		if err != nil {
-			return detailedRollout, err
-		}
-		detailedRollout.Definition.Strategy = strategy
+		wg.Add(1)
+		go func(wg *sync.WaitGroup, errors chan error) {
```
Comment (caduri) [inline viewcontroller/viewcontroller.go:111]: add an empty line

File: viewcontroller/viewcontroller.go [hunk 4]
```diff
@@ -108,48 +100,61 @@ func NewRolloutViewController(rolloutId string, noColor bool) *RolloutViewContro
...
+			defer wg.Done()
+			strategy, err := c.rolloutRepository.GetStrategy(c.rolloutId)
```
Comment (caduri) [inline viewcontroller/viewcontroller.go:113]: add an empty line, please format the code with empty lines
you know where I like them

File: viewcontroller/viewcontroller.go [hunk 5]
```diff
@@ -108,48 +100,61 @@ func NewRolloutViewController(rolloutId string, noColor bool) *RolloutViewContro
...
-	fetchedRollout, err := oceancd.GetRollout(c.rolloutId)
-	if err != nil {
-		return detailedRollout, err
-	}
-
-	phases, err := oceancd.GetRolloutPhases(c.rolloutId)
-	if err != nil {
-		return detailedRollout, err
-	}
+	go func(wg *sync.WaitGroup, errors chan error) {
```
Comment (caduri) [inline viewcontroller/viewcontroller.go:123]: can we extract these anonymous functions into methods with names so we can understand what they should do?

**Review by caduri — CHANGES_REQUESTED** [review]

File: viewcontroller/viewcontroller.go (hunk 5 above)
Comment (konstantin-lysenko-netapp) [reply viewcontroller/viewcontroller.go:123]: yes

File: viewcontroller/viewcontroller.go (hunk 2 above)
Comment (konstantin-lysenko-netapp) [reply viewcontroller/viewcontroller.go:107]: 3 means that runner will be blocked by the wg.Wait() until 3 goroutines will call wg.Done()

File: viewcontroller/viewcontroller.go (hunk 4 above)
Comment (konstantin-lysenko-netapp) [reply viewcontroller/viewcontroller.go:113]: please specify places, I remember only about separating if each from other

File: viewcontroller/viewcontroller.go (hunk 2 above)
Comment (caduri) [reply viewcontroller/viewcontroller.go:108]: which go routines?

File: viewcontroller/viewcontroller.go (hunk 5 above)
Comment (caduri) [reply viewcontroller/viewcontroller.go:125]: lets do it :)

File: viewcontroller/viewcontroller.go (hunk 2 above)
Comment (caduri) [reply viewcontroller/viewcontroller.go:108]: I think I need an explanation regarding this, we will go over it today

**Review by caduri — CHANGES_REQUESTED** [review]

File: README.md [hunk 6]
```diff
@@ -62,9 +62,9 @@ To build Ocean CD CLI from the source you have to:
     chmod +x /usr/local/bin/oceancd
     ```
 
-3. Add the `/user/local/bin` path if the `PATH` variable doesn't contain it:
+3. Add the `/usr/local/bin` path if the `PATH` variable doesn't contain it:
     ```
-    PATH=$PATH:/user/local/bin
+    PATH=$PATH:/usr/local/bin
```
Comment (caduri) [inline README.md:67]: good catch

File: pkg/oceancd/builders/rollout.go [hunk 7]
```diff
@@ -0,0 +1,94 @@
...
+
+	b.wg = &sync.WaitGroup{}
+
+	if b.withStrategy {
+		//add setStrategy()
+		b.wg.Add(1)
+		go b.setStrategy(rolloutId)
+	}
+
+	//add setRollout()
```
Comment (caduri) [inline pkg/oceancd/builders/rollout.go:36]: mention that this group waits to go routine of `setStrategy`
It's a comment don't be shy with words :)

Comment (caduri) [reply pkg/oceancd/builders/rollout.go:36]: do the same to the other comments

File: pkg/oceancd/builders/rollout.go [hunk 8]
```diff
@@ -0,0 +1,94 @@
...
+	default:
+		return b.detailedRollout, nil
+	}
+}
+
+func (b *DetailedRolloutBuilder) setStrategy(rolloutId string) {
+	defer b.wg.Done()
+	strategy, err := b.rolloutRepository.GetStrategy(rolloutId)
+	b.detailedRollout.Definition.Strategy = strategy
+	if err != nil {
```
Comment (caduri) [inline pkg/oceancd/builders/rollout.go:59]: empty line

File: pkg/oceancd/builders/rollout.go [hunk 9]
```diff
@@ -0,0 +1,94 @@
...
+		b.errors <- err
+	}
+}
+
+func (b *DetailedRolloutBuilder) setRollout(rolloutId string) {
+	defer b.wg.Done()
+	fetchedRollout, err := oceancd.GetRollout(rolloutId)
+	b.detailedRollout.Rollout = fetchedRollout
+	if err != nil {
```
Comment (caduri) [inline pkg/oceancd/builders/rollout.go:68]: empty line

**Review by caduri — CHANGES_REQUESTED** [review]

File: pkg/oceancd/builders/rollout.go [hunk 10]
```diff
@@ -0,0 +1,94 @@
...
+		b.errors <- err
+	}
+}
+
+func (b *DetailedRolloutBuilder) setRolloutVerifications(rolloutId string) {
+	defer b.wg.Done()
+	verifications, err := oceancd.GetRolloutVerifications(rolloutId)
+	b.detailedRollout.Verifications = verifications
+	if err != nil {
```
[the remark this answers is not in the document]
Comment (konstantin-lysenko-netapp) [reply pkg/oceancd/builders/rollout.go:86]: do you want to add an empty line before the if statement? usually, an empty line isn't being added in this case

File: pkg/oceancd/builders/rollout.go (hunk 8 above)
Comment (caduri) [reply pkg/oceancd/builders/rollout.go:58]: you missed this one

File: pkg/oceancd/builders/rollout.go (hunk 9 above)
Comment (caduri) [reply pkg/oceancd/builders/rollout.go:68]: you missed this one

File: pkg/oceancd/builders/rollout.go (hunk 10 above)
[the remark this answers is not in the document]
Comment (caduri) [reply pkg/oceancd/builders/rollout.go:88]: this is part of spot coding standards

**Review by caduri — CHANGES_REQUESTED** [review]

**Review by caduri — APPROVED** [review]

**Pull request merged**